hbase-common/src/test/java/org/apache/hadoop/hbase/util/TestBytes.java

```diff
@@ -58,7 +59,7 @@ private static void setUnsafe(boolean value) throws Exception {
     } finally {
       modifiersField.setInt(field, oldModifiers);
     }
-    assertEquals(Bytes.UNSAFE_UNALIGNED, value);
+    assertEquals(UnsafeAvailChecker.unaligned(), value);
   }
 
```

Comment:
While we are here, and this is not a requirement (I've already approved the PR), would be good to have a small new unit test that confirms the new configuration parameter does in fact disable Unsafe usage when we would otherwise use it.

Reply:
@apurtell Added a test in the new commit. Added a resource in the classpath for the class to consume the flag during class loading in the static block. Please review and let me know your thoughts